nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,783 @@
+
+try:
+    from urllib.parse import quote
+except ImportError:
+    from urllib import quote  # Python2
+
+try:
+    from html import escape
+except ImportError:
+    from cgi import escape
+
+import pandas as pd
+
+from matplotlib import pyplot as plt
+from nilearn.plotting import (plot_glass_brain,
+                              plot_roi,
+                              plot_stat_map,
+                              )
+from nilearn.plotting.img_plotting import MNI152TEMPLATE
+from nilearn.plotting.js_plotting_utils import HTMLDocument
+
+from nistats.reporting import (plot_contrast_matrix,
+                               plot_design_matrix,
+                               get_clusters_table,
+                               )
+from nistats.thresholding import map_threshold
+
+html_template_root_path = os.path.join(os.path.dirname(__file__),
+                                       'glm_reporter_templates')
+
+
+def make_glm_report(
+        model,
+        contrasts,
+        title='auto',
+        roi_img=None,
+        bg_img=MNI152TEMPLATE,
+        threshold=3.09,
+        alpha=0.01,
+        cluster_threshold=0,
+        height_control='fpr',
+        min_distance=8.,
+        plot_type='slice',
+        display_mode=None,
+        nb_width=1600,
+        nb_height=800,
+        ):
+    """ Returns HTMLDocument object for a report which shows all important aspects of a fitted GLM. The object can be opened in a browser, displayed in a notebook, or saved to disk as a portable HTML file.
+    
+    Usage:
+        report = make_glm_report(model, contrasts)
+        report.open_in_browser()
+        report.save_as_html(destination_path)
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        Must have the computed design matrix(ces).
+        
+    contrasts: Dict[string, ndarray] , String, List[ndarrays]
+        Contrasts information for a first or second level model.
+        Corresponds to the ``contrast_def`` for the FirstLevelModel [1]_
+        & second_level_contrast for a SecondLevelModel [2]_ .
+    
+    title: String, default 'auto'
+        Text to represent the web page's title and primary heading.
+        If 'auto', uses the contrast titles to generate a title.
+        
+    roi_img : Niimg-like object
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        The ROI/mask image, it could be binary mask or an atlas or ROIs
+        with integer values.
+    
+    bg_img : Niimg-like object
+        Default is the MNI152 template
+        See http://nilearn.github.io/manipulating_images/input_output.html
+        The background image that the ROI/mask will be plotted on top of.
+        To turn off background image, just pass "bg_img=None".
+        
+    threshold: float
+        Default is 3.09
+        Cluster forming threshold in same scale as `stat_img` (either a
+        p-value or z-scale value).
```

Comment:
t-scale or z-scale value, not p-value

Reply:
In the other docstrings it's z-scale. Should I only mention z-scale?

Reply:
Yes, only z-scale.